x-pack/filebeat/input/azureblobstorage/job.go

```diff
@@ -161,53 +321,35 @@ func (j *job) addGzipDecoderIfNeeded(body io.Reader) (io.Reader, error) {
 	return gzip.NewReader(bufReader)
 }
 
-func (j *job) readJsonAndPublish(ctx context.Context, r io.Reader, id string) error {
-	dec := json.NewDecoder(r)
-	dec.UseNumber()
-	var offset int64
-	var relativeOffset int64
-	// uncompressed files use the client to directly set the offset, this
-	// in turn causes the offset to reset to 0 for the new stream, hence why
-	// we need to keep relative offsets to keep track of the actual offset
-	if !j.isCompressed {
-		relativeOffset = j.offset
-	}
-	for dec.More() && ctx.Err() == nil {
-		var item json.RawMessage
-		offset = dec.InputOffset()
-		if err := dec.Decode(&item); err != nil {
-			return fmt.Errorf("failed to decode json: %w", err)
-		}
-		// manually seek offset only if file is compressed
-		if j.isCompressed && offset < j.offset {
-			continue
+// evaluateJSON, uses a bufio.NewReader & reader.Peek to evaluate if the
+// data stream contains a json array as the root element or not, without
+// advancing the reader. If the data stream contains an array as the root
+// element, the value of the boolean return type is set to true.
+func evaluateJSON(reader *bufio.Reader) (io.Reader, bool, error) {
+	eof := false
+	for i := 0; ; i++ {
+		b, err := reader.Peek((i + 1) * 5)
```

Comment:
What is 5?

Reply:
This logic was copied from the gcs input. If I remember correctly, I think this choice for the value **5** was arbitrary, basically tries to read in incremental small chunks. This value worked well for the use case hence never changed it.

Reply:
Let's have a constant with a comment explaining this.